pokemon_battle_arena/src/arena/mod.rs

```diff
@@ -29,18 +28,22 @@ impl Arena {
     // Getter Methods
     //
     /// Gets the type of the arena
-    pub fn get_effect(&self) -> enums::types {
+    #[allow(dead_code)]
+    pub fn get_effect(&self) -> enums::Types {
         self.effect.clone()
     }
     /// Gets the actual weather of the arena
+    #[allow(dead_code)]
     pub fn get_weather(&self) -> enums::Weather {
         self.weather.clone()
     }
     /// Returns a player one
+    #[allow(dead_code)]
     pub fn get_player_one(&self) -> Player {
         self.player_1.clone()

```

Comment:
Or you could just return a reference to a `Player`. This avoid cloning, too. You should never `clone()` just to satisfy the borrow checker.